cloudinit/config/cc_ubuntu_autoinstall.py

```diff
@@ -0,0 +1,141 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+
+"""Autoinstall: Support ubuntu live-server autoinstall syntax."""
+
+import re
+from textwrap import dedent
+
+from cloudinit import log as logging
+from cloudinit.config.schema import (
+    MetaSchema,
+    SchemaProblem,
+    SchemaValidationError,
+    get_meta_doc,
+)
+from cloudinit.settings import PER_INSTANCE
+from cloudinit.subp import subp
+
+LOG = logging.getLogger(__name__)
+
+distros = ["ubuntu"]
+
+meta: MetaSchema = {
+    "id": "cc_ubuntu_autoinstall",
+    "name": "Autoinstall",
+    "title": "Support Ubuntu live-server install syntax",
+    "description": dedent(
+        """\
+        Ubuntu's autoinstall syntax supports single-system automated installs
+        in either the live-server or live-desktop installers.
+        When "autoinstall" directives are provided in either
+        #cloud-config user-data or ``/etc/cloud/cloud.cfg.d`` validate
+        minimal autoinstall schema adherance and emit a warning if the
+        live-installer is not present.
+
+        The live-installer will use autoinstall directives to seed answers to
+        configuration prompts during system install to allow for a
+        "touchless" Ubuntu system install.
+
+        For more details on Ubuntu's autoinstaller:
+            https://ubuntu.com/server/docs/install/autoinstall
+    """
+    ),
+    "distros": distros,
+    "examples": [
+        dedent(
+            """\
+            # Tell the live-server installer to provide dhcp6 network config
+            # and LVM on a disk matching the serial number prefix CT
+            autoinstall:
+              version: 1
+              network:
+                version: 2
+                ethernets:
+                  enp0s31f6:
+                    dhcp6: yes
+              storage:
+                layout:
+                  name: lvm
+                  match:
+                    serial: CT*
+        """
+        )
+    ],
+    "frequency": PER_INSTANCE,
```

Comment:
`ONCE`?

Reply:
@blackboxsw Did you see my frequency comment here? Should we do ONCE instead of PER_INSTANCE?